examples/hack_acq_params.py

```diff
@@ -0,0 +1,259 @@
+"""
+Hack H5 Properties
+==================
+
+There are various labeling properties that are stored in the HDF5 file.
+Sometimes we mess up, and we need to hack these properties ex-post-facto.
+
+Note that is intended *only* for hacking labeling/categorizing properties, and
+should *never* be used to hack information about pulse program parameters, etc,
+that were actually sent to the spectrometer.
+
+The GPT conversation is `here <https://chatgpt.com/c/66f5aef7-8f68-800b-8a13-0b133ede1ca5>`_.
+"""
+import sys
+import h5py
+from PyQt5.QtWidgets import (
+    QApplication,
+    QWidget,
+    QVBoxLayout,
+    QLabel,
+    QLineEdit,
+    QPushButton,
+    QMessageBox,
+    QHBoxLayout,
+)
+from pyspecdata import search_filename
+
+
+class NodeAsDict:
+    def __init__(self, node):
+        self.node = node
+
+    def __getitem__(self, key):
+        if key in self.node.attrs:
+            value = self.node.attrs[key]
+            if isinstance(value, bytes):
+                return value.decode(
+                    "utf-8"
+                )  # Decode byte string to UTF-8 string
+            return value
+        elif key in self.node:
+            return NodeAsDict(
+                self.node[key]
+            )  # Return a group as another NodeAsDict instance
+        else:
+            return None
+
+    def __setitem__(self, key, value):
+        if isinstance(value, (int, float)):  # Numbers stored as attributes
+            self.node.attrs[key] = value
+        elif isinstance(
+            value, str
+        ):  # Strings stored as byte strings (UTF-8 encoded)
+            self.node.attrs[key] = value.encode("utf-8")
+        elif isinstance(value, dict):  # Dictionaries stored as groups
+            if key not in self.node:
+                self.node.create_group(key)  # Create group if it doesn't exist
+            group = self.node[key]
+            node_dict = NodeAsDict(group)
+            for subkey, subvalue in value.items():
+                node_dict[
+                    subkey
+                ] = subvalue  # Recursively assign subkeys and values
+        elif isinstance(value,bytes):
+            self.node.attrs[key] = value.decode( "utf-8")
```

Comment:
below in save_changes (line 226) we convert to bytes so this is needed to decode